src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -977,20 +989,63 @@ private void onDecodePublish(
+            final String16FW publishTopicName = publish.topicName();
+            final DirectBuffer buffer = publishTopicName.buffer();
 
-            String info = "info";
+            OctetsFW properties = publish.properties();
+            final int propertiesOffset = properties.offset();
+            final int propertiesLimit = properties.limit();
 
-            final MqttDataExFW dataEx = mqttDataExRW
-                                            .wrap(dataExtBuffer, 0, dataExtBuffer.capacity())
-                                            .typeId(mqttTypeId)
-                                            .topic(topicName)
-                                            .expiryInterval(15)
-                                            .contentType("message")
-                                            .format(f -> f.set(MqttPayloadFormat.TEXT))
-                                            .responseTopic(topicName)
-                                            .correlationInfo(c -> c.bytes(b -> b.set(info.getBytes(UTF_8))))
-                                            .build();
+            final String topicName = publishTopicName.asString();
+
+            MqttPayloadFormat mqttPayloadFormat = MqttPayloadFormat.TEXT;
+            int messageExpiryInterval = 0;
+            String contentType = "";
+            String responseTopic = "";
+            OctetsFW correlationInfo = EMPTY_OCTETS;
+
+            MqttPropertyFW mqttProperty;
+
+            for (int progress = propertiesOffset; progress < propertiesLimit; progress = mqttProperty.limit())
+            {
+                mqttProperty = mqttPropertyRO.tryWrap(buffer, progress, propertiesLimit);
+                switch (mqttProperty.kind())
+                {
+                case KIND_PAYLOAD_FORMAT_INDICATOR:
+                    mqttPayloadFormat = MqttPayloadFormat.valueOf(mqttProperty.payloadFormatIndicator());
+                    break;
+                case KIND_MESSAGE_EXPIRY_INTERVAL:
+                    messageExpiryInterval = mqttProperty.messageExpiryInterval();
+                    break;
+                case KIND_CONTENT_TYPE:
+                    contentType = mqttProperty.contentType().asString();
+                    break;
+                case KIND_RESPONSE_TOPIC:
+                    responseTopic = mqttProperty.responseTopic().asString();
+                    break;
+                case KIND_CORRELATION_DATA:
+                    correlationInfo = mqttProperty.correlationData().bytes();
+                    break;
+                case KIND_TOPIC_ALIAS:
+                    break;
+                default:
+                    onDecodeError(traceId, authorization, MALFORMED_PACKET);
+                    break;
+                }
+            }
+
+            final MqttPayloadFormat format = mqttPayloadFormat != null ? mqttPayloadFormat : MqttPayloadFormat.TEXT;
+            final OctetsFW finalCorrelationInfo = correlationInfo;
```

Comment:
In these cases, where we need to make a non-final variable final for the sake of a lambda, we typically just use the same variable name and add a `0` at the end, so `mqttPayloadFormat0` and `correlationInfo0`.